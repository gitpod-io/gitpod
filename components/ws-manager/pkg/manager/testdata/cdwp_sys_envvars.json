{
  "spec": {
      "ideImage": {
          "webRef": "eu.gcr.io/gitpod-core-dev/buid/theia-ide:someversion"
      },
      "workspaceImage": "some-ref",
      "initializer": {
          "snapshot": {
              "snapshot": "workspaces/cryptic-id-goes-herg/fd62804b-4cab-11e9-843a-4e645373048e.tar@gitpod-dev-user-christesting"
          }
      },
      "envvars": [
          { "name": "GITPOD_WORKSPACE_CONTEXT", "value": "will make it" },
          { "name": "GITPOD_WORKSPACE_CONTEXT_URL", "value": "will make it" },
          { "name": "GITPOD_WORKSPACE_CLASS_INFO", "value": "will make it" },
          { "name": "GITPOD_TASKS", "value": "will make it" },
          { "name": "GITPOD_RESOLVED_EXTENSIONS", "value": "will make it" },
          { "name": "GITPOD_EXTERNAL_EXTENSIONS", "value": "will make it" },
          { "name": "GITPOD_IDE_ALIAS", "value": "will make it" },

          { "name": "GITPOD_SOMETHING_ELSE", "value": "will NOT make it"},

          { "name": "something_without_gitpod", "value":"will make it" },
          { "name": "one_from_a_secret", "secret": { "secretName": "some-secret", "key": "some-key" } }
      ],
      "sysEnvvars": [
          { "name": "GITPOD_SYSTEM_VAR", "value": "will make it"},
          { "name": "GITPOD_IDE_ALIAS", "value": "should be override" }
      ],
      "git": {
          "username": "usernameGoesHere",
          "email": "some@user.com"
      }
  }
}
