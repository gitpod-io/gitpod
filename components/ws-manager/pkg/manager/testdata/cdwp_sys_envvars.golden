{
    "reason": {
        "metadata": {
            "name": "ws-test",
            "namespace": "default",
            "creationTimestamp": null,
            "labels": {
                "app": "gitpod",
                "component": "workspace",
                "gitpod.io/networkpolicy": "default",
                "gitpod.io/workspaceClass": "default",
                "gpwsman": "true",
                "headless": "false",
                "metaID": "foobar",
                "owner": "tester",
                "project": "gitpod",
                "team": "awesome",
                "workspaceID": "test",
                "workspaceType": "regular"
            },
            "annotations": {
                "cluster-autoscaler.kubernetes.io/safe-to-evict": "false",
                "container.apparmor.security.beta.kubernetes.io/workspace": "unconfined",
                "gitpod.io/attemptingToCreate": "true",
                "gitpod/admission": "admit_owner_only",
                "gitpod/contentInitializer": "GmcKZXdvcmtzcGFjZXMvY3J5cHRpYy1pZC1nb2VzLWhlcmcvZmQ2MjgwNGItNGNhYi0xMWU5LTg0M2EtNGU2NDUzNzMwNDhlLnRhckBnaXRwb2QtZGV2LXVzZXItY2hyaXN0ZXN0aW5n",
                "gitpod/id": "test",
                "gitpod/imageSpec": "Cghzb21lLXJlZhI0ZXUuZ2NyLmlvL2dpdHBvZC1jb3JlLWRldi9idWlkL3RoZWlhLWlkZTpzb21ldmVyc2lvbjoAOgA=",
                "gitpod/never-ready": "true",
                "gitpod/ownerToken": "%7J'[Of/8NDiWE+9F,I6^Jcj_1\u0026}-F8p",
                "gitpod/servicePrefix": "foobarservice",
                "gitpod/url": "test-foobarservice-gitpod.io"
            },
            "finalizers": [
                "gitpod.io/finalizer"
            ]
        },
        "spec": {
            "volumes": [
                {
                    "name": "vol-this-workspace",
                    "hostPath": {
                        "path": "/tmp/workspaces/test",
                        "type": "DirectoryOrCreate"
                    }
                },
                {
                    "name": "daemon-mount",
                    "hostPath": {
                        "path": "/tmp/workspaces/test-daemon",
                        "type": "DirectoryOrCreate"
                    }
                }
            ],
            "containers": [
                {
                    "name": "workspace",
                    "image": "registry-facade:8080/remote/test",
                    "command": [
                        "/.supervisor/workspacekit",
                        "ring0"
                    ],
                    "ports": [
                        {
                            "containerPort": 23000
                        },
                        {
                            "name": "supervisor",
                            "containerPort": 22999
                        }
                    ],
                    "env": [
                        {
                            "name": "GITPOD_REPO_ROOT",
                            "value": "/workspace"
                        },
                        {
                            "name": "GITPOD_REPO_ROOTS",
                            "value": "/workspace"
                        },
                        {
                            "name": "GITPOD_CLI_APITOKEN",
                            "value": "Ab=5=rRA*9:C'T{;RRB\u003e]vK2p6`fFfrS"
                        },
                        {
                            "name": "GITPOD_OWNER_ID",
                            "value": "tester"
                        },
                        {
                            "name": "GITPOD_WORKSPACE_ID",
                            "value": "foobar"
                        },
                        {
                            "name": "GITPOD_INSTANCE_ID",
                            "value": "test"
                        },
                        {
                            "name": "GITPOD_THEIA_PORT",
                            "value": "23000"
                        },
                        {
                            "name": "THEIA_WORKSPACE_ROOT",
                            "value": "/workspace"
                        },
                        {
                            "name": "GITPOD_HOST",
                            "value": "gitpod.io"
                        },
                        {
                            "name": "GITPOD_WORKSPACE_URL",
                            "value": "test-foobarservice-gitpod.io"
                        },
                        {
                            "name": "THEIA_SUPERVISOR_ENDPOINT",
                            "value": ":22999"
                        },
                        {
                            "name": "THEIA_WEBVIEW_EXTERNAL_ENDPOINT",
                            "value": "webview-{{hostname}}"
                        },
                        {
                            "name": "THEIA_MINI_BROWSER_HOST_PATTERN",
                            "value": "browser-{{hostname}}"
                        },
                        {
                            "name": "GITPOD_GIT_USER_NAME",
                            "value": "usernameGoesHere"
                        },
                        {
                            "name": "GITPOD_GIT_USER_EMAIL",
                            "value": "some@user.com"
                        },
                        {
                            "name": "GITPOD_SYSTEM_VAR",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_IDE_ALIAS",
                            "value": "should be override"
                        },
                        {
                            "name": "GITPOD_WORKSPACE_CONTEXT",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_WORKSPACE_CONTEXT_URL",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_WORKSPACE_CLASS_INFO",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_TASKS",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_RESOLVED_EXTENSIONS",
                            "value": "will make it"
                        },
                        {
                            "name": "GITPOD_EXTERNAL_EXTENSIONS",
                            "value": "will make it"
                        },
                        {
                            "name": "something_without_gitpod",
                            "value": "will make it"
                        },
                        {
                            "name": "one_from_a_secret",
                            "valueFrom": {
                                "secretKeyRef": {
                                    "name": "some-secret",
                                    "key": "some-key"
                                }
                            }
                        },
                        {
                            "name": "GITPOD_INTERVAL",
                            "value": "30000"
                        },
                        {
                            "name": "GITPOD_MEMORY",
                            "value": "999"
                        }
                    ],
                    "resources": {
                        "limits": {
                            "cpu": "900m",
                            "memory": "1G"
                        },
                        "requests": {
                            "cpu": "899m",
                            "ephemeral-storage": "5Gi",
                            "memory": "999M"
                        }
                    },
                    "volumeMounts": [
                        {
                            "name": "vol-this-workspace",
                            "mountPath": "/workspace",
                            "mountPropagation": "HostToContainer"
                        },
                        {
                            "name": "daemon-mount",
                            "mountPath": "/.workspace",
                            "mountPropagation": "HostToContainer"
                        }
                    ],
                    "readinessProbe": {
                        "httpGet": {
                            "path": "/_supervisor/v1/status/content/wait/true",
                            "port": 22999,
                            "scheme": "HTTP"
                        },
                        "initialDelaySeconds": 2,
                        "timeoutSeconds": 1,
                        "periodSeconds": 1,
                        "successThreshold": 1,
                        "failureThreshold": 600
                    },
                    "terminationMessagePolicy": "File",
                    "imagePullPolicy": "IfNotPresent",
                    "securityContext": {
                        "capabilities": {
                            "add": [
                                "AUDIT_WRITE",
                                "FSETID",
                                "KILL",
                                "NET_BIND_SERVICE",
                                "SYS_PTRACE"
                            ],
                            "drop": [
                                "SETPCAP",
                                "CHOWN",
                                "NET_RAW",
                                "DAC_OVERRIDE",
                                "FOWNER",
                                "SYS_CHROOT",
                                "SETFCAP",
                                "SETUID",
                                "SETGID"
                            ]
                        },
                        "privileged": false,
                        "runAsUser": 33333,
                        "runAsGroup": 33333,
                        "runAsNonRoot": true,
                        "readOnlyRootFilesystem": false,
                        "allowPrivilegeEscalation": true
                    }
                }
            ],
            "restartPolicy": "Never",
            "serviceAccountName": "workspace",
            "automountServiceAccountToken": false,
            "securityContext": {
                "seccompProfile": {
                    "type": "Localhost",
                    "localhostProfile": "workspace-default"
                }
            },
            "hostname": "foobar",
            "affinity": {
                "nodeAffinity": {
                    "requiredDuringSchedulingIgnoredDuringExecution": {
                        "nodeSelectorTerms": [
                            {
                                "matchExpressions": [
                                    {
                                        "key": "gitpod.io/workload_workspace_regular",
                                        "operator": "Exists"
                                    },
                                    {
                                        "key": "gitpod.io/ws-daemon_ready_ns_default",
                                        "operator": "Exists"
                                    },
                                    {
                                        "key": "gitpod.io/registry-facade_ready_ns_default",
                                        "operator": "Exists"
                                    }
                                ]
                            }
                        ]
                    }
                }
            },
            "tolerations": [
                {
                    "key": "node.kubernetes.io/disk-pressure",
                    "operator": "Exists",
                    "effect": "NoExecute"
                },
                {
                    "key": "node.kubernetes.io/memory-pressure",
                    "operator": "Exists",
                    "effect": "NoExecute"
                },
                {
                    "key": "node.kubernetes.io/network-unavailable",
                    "operator": "Exists",
                    "effect": "NoExecute",
                    "tolerationSeconds": 30
                }
            ],
            "enableServiceLinks": false
        },
        "status": {}
    }
}